### PR TITLE
Basic Nix compatibility

### PR DESCRIPTION
**Pull request by qsantos**
This PR extracts 2 changes from https://github.com/helsing-ai/buffrs/pull/172 that should be safe but should help in making Buffrs compatible with Nix.

### CODE REVIEW

**Review by j-baker — APPROVED** [review]

File: src/credentials.rs [hunk 1]
```diff
@@ -105,17 +105,12 @@ impl Credentials {
...
+    /// not be performing global stateful operations in absence of a user instruction.
     pub async fn load() -> miette::Result<Self> {
-        let Some(credentials) = Self::read().await? else {
-            let credentials = Credentials::default();
-            credentials.write().await.wrap_err("Writing empty config")?;
-            return Ok(credentials);
-        };
-
-        Ok(credentials)
+        Ok(Self::read().await?.unwrap_or_else(Credentials::default))
```
Comment (mara-schulke) [inline src/credentials.rs:113]: huh? why should we want this?

**Comment by mara-schulke** [comment]
So as far as I understand your issue originates from buffrs home being non-existent in the nix world?

File: src/credentials.rs (hunk 1 above)
Comment (j-baker) [reply src/credentials.rs:113]: maybe put differently - isn't it strange for a build.rs to create files in $HOME?

**Review by mara-schulke — APPROVED** [review]

**Comment by qsantos** [comment]
What's going on with `typos`?

**Comment by qsantos** [comment]
Thanks @BlowaXD!

**Pull request merged**